pkg/data/account.go

```diff
@@ -0,0 +1,146 @@
+package data
+
+import (
+	gErrors "errors"
+	"fmt"
+	"strconv"
+
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/Optum/dce/pkg/model"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+)
+
+// Account - Data Layer Struct
+type Account struct {
+	DynamoDB       dynamodbiface.DynamoDBAPI
+	TableName      string `env:"ACCOUNT_DB"`
+	ConsistentRead bool   `env:"USE_CONSISTENT_READS" envDefault:"false"`
+}
+
+// WriteAccount the Account record in DynamoDB
+func (a *Account) WriteAccount(account *model.Account, lastModifiedOn *int64) error {
+
+	var expr expression.Expression
+	var err error
+	var returnValue string
+	// lastModifiedOn is nil on a create
+	if lastModifiedOn != nil {
+		modExpr := expression.Name("LastModifiedOn").Equal(expression.Value(lastModifiedOn))
+		expr, err = expression.NewBuilder().WithCondition(modExpr).Build()
+		if err != nil {
+			return errors.NewInternalServer("error building query", err)
+		}
+		returnValue = "NONE"
+	} else {
+		modExpr := expression.Name("LastModifiedOn").AttributeNotExists()
+		expr, err = expression.NewBuilder().WithCondition(modExpr).Build()
+		if err != nil {
+			return errors.NewInternalServer("error building query", err)
+		}
+		returnValue = "NONE"
```

Comment:
Looks like this is always set to `"NONE"`?

Reply:
On that note -- I'm thinking a `POST /accounts` or `PUT /accounts` request should return the new account object, so maybe this needs to return `"ALL_NEW"` in order to populate that object.

Reply:
or, the controller needs to run an extra query, to grab the account info.

Reply:
The current logic works with NONE.  The big debate is around the patch or DynamoDBs UpdateItem vs PutItem.  Right now we use PutItem which means we have to have the complete record before the update.